docs/ca/dev/monitoring/opentelemetry-instrumentation.md

```diff
@@ -0,0 +1,437 @@
+                array $params,
+                $returnValue, //Result of the method execution.
+                ?\Throwable $exception //Exception if one was thrown during the execution
+                ) {
+                    $scope = \OpenTelemetry\Context\Context::storage()->scope();
+
+                    if (null === $scope) {
+                        return;
+                    }
+
+                    //Here you can just check the $exception value. But in some cases you might want to take if from other places. E.g. if method that thrown exception was not instrumented, we still want to catch it in a parent span.
+                    $error = error_get_last();
+
+                    if (is_array($error) && in_array($error['type'], [E_ERROR, E_CORE_ERROR, E_COMPILE_ERROR, E_PARSE], true)) {
+                        $exception = new \Exception(
+                            'Error: ' . $error['message'] . ' in ' . $error['file'] . ' on line ' . $error['line']
+                        );
+                    }
+
+                    $scope->detach();
+                    $span = \Spryker\Service\Opentelemetry\Instrumentation\Span\Span::fromContext($scope->context());
+
+                    if ($exception !== null) {
+                        $span->recordException($exception);//Exception will be attached as an event into the span.
+                        $span->setAttribute('error_message', $exception->getMessage());
+                        $span->setAttribute('error_code', $exception->getCode());
+                    }
+
+                    //Status code adds some visibility. Error status code will mark your span as a one with an error for easier navigation.
+                    $span->setStatus($exception !== null ? \OpenTelemetry\API\Trace\StatusCode::STATUS_ERROR : \OpenTelemetry\API\Trace\StatusCode::STATUS_OK);
+
+                    //Span ends and sent into a span processor to be validated and prepared for exporting.
+                    $span->end();
+                }
+            );
+```
+
+</details>
+
+## Collector
+Collector collects traces and sends them to your monitoring platform. Sending traces to collector is done after that request is sent so it doesn't impact response time. Collector is not part of the application and should be prepared by a Cloud engineer or you can add one for you local setup yourself.
+
+## Integration
+
+Run the latest version of the script from the [Installer repo](https://github.com/spryker/opentelemetry-installer/).
+
+If you want to integrate manually, the following sections describe all steps of the scripts.
+
+### Install required packages
+OpenTelemetry provides instrumentation via packages that can be installed to register hooks automatically. If you want to instrument something that is not covered in our code base but is required for you, you can try to install one of the packages listed at [Registry](https://opentelemetry.io/ecosystem/registry/?language=php) or from other sources.
```

Comment:
What do you mean by `something`?

Reply:
Anything from 3rd party systems. E.g. Symfony packages